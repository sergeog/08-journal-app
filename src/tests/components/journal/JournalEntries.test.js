import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';

import notesMock from '../../fixtures/notesMock';
import { JournalEntries } from '../../../components/journal/JournalEntries';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  notes: {
    notes: notesMock,
  },
};
const store = mockStore(initialState);

const wrapper = mount(
  <Provider store={store}>
    <JournalEntries />
  </Provider>
);

describe('Pruebas en <JournalEntries />', () => {
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar los entries', () => {
    const journalEntryTag = wrapper.find('JournalEntry');
    expect(journalEntryTag.exists()).toBe(true);
  });
});
