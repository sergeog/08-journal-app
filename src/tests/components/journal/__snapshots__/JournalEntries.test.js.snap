// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntries /> debe mostrarse correctamente 1`] = `
<div
  className="journal__entries"
>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 1
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 1
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        19th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-picture"
      style={
        Object {
          "backgroundImage": "url(https://res.cloudinary.com/dbeq1o9b1/image/upload/v1608408608/ce8qbbh5njfhkhboes0p.png)",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 2
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 2
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        12th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-picture"
      style={
        Object {
          "backgroundImage": "url(https://res.cloudinary.com/dbeq1o9b1/image/upload/v1608416967/zasq87yoclsuss8sv0cz.png)",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 3
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 3
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        12th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 4
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 4
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        12th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 5
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 5
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        19th
      </h4>
    </div>
  </div>
  <div
    className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        Title de la Nota 6
      </p>
      <p
        className="journal__entry-content"
      >
        Body de la Nota 6
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Sunday
      </span>
      <h4>
        13th
      </h4>
    </div>
  </div>
</div>
`;
